src/test/ui/constructor-lifetime-args.stderr

```diff
@@ -1,26 +1,66 @@
-error[E0107]: wrong number of lifetime arguments: expected 2, found 1
+error[E0107]: this struct takes 2 lifetime arguments but only 1 lifetime argument was supplied
   --> $DIR/constructor-lifetime-args.rs:17:5
    |
 LL |     S::<'static>(&0, &0);
-   |     ^^^^^^^^^^^^ expected 2 lifetime arguments
+   |     ^   ------- supplied 1 lifetime argument
+   |     |
+   |     expected 2 lifetime arguments
+   |
+note: struct defined here, with 2 lifetime parameters: `'a`, `'b`
+  --> $DIR/constructor-lifetime-args.rs:9:8
+   |
+LL | struct S<'a, 'b>(&'a u8, &'b u8);
+   |        ^ --  --
+help: add missing lifetime argument
+   |
+LL |     S::<'static, 'b>(&0, &0);
+   |                ^^^^
```

Comment:
For lifetimes we would ideally look for available lifetimes (like we do in other cases) and if none use `'static`, but again, that should be follow up work. (I'm just marking TODO notes for later right now.)

Reply:
Parallel idea: for lifetime & type generics in _expression position_ (e.g. `.collect::<Vec>()`) we could suggest using placeholders (i.e. suggest `Vec<_>` instead of `Vec<T>`).